### PR TITLE
implement a no_store option for pki roles, refs #2511

### PR DESCRIPTION
**Pull request by slingamn**
This implements option 1 from #2511. The implementation (including the test) is mostly cargo-culted from the implementation of `generate_lease`. I didn't update `website/source/api` because that looked like it might be autogenerated?

Thanks for your time.

### CODE REVIEW

**Comment by jefferai** [comment]
>  I didn't update website/source/api because that looked like it might be autogenerated?

It's not -- please add it!

File: builtin/logical/pki/path_issue_sign.go [hunk 1]
```diff
@@ -240,10 +240,12 @@ func (b *backend) pathIssueSignCert(
 		resp.Secret.TTL = parsedBundle.Certificate.NotAfter.Sub(time.Now())
 	}
 
-	err = req.Storage.Put(&logical.StorageEntry{
-		Key:   "certs/" + cb.SerialNumber,
-		Value: parsedBundle.CertificateBytes,
-	})
+	if !*role.NoStore {
```
Comment (jefferai) [inline builtin/logical/pki/path_issue_sign.go]: No need to dereference the role here.

File: builtin/logical/pki/path_roles.go [hunk 2]
```diff
@@ -504,6 +529,7 @@ type roleEntry struct {
 	OU                    string `json:"ou" structs:"ou" mapstructure:"ou"`
 	Organization          string `json:"organization" structs:"organization" mapstructure:"organization"`
 	GenerateLease         *bool  `json:"generate_lease,omitempty" structs:"generate_lease,omitempty"`
+	NoStore               *bool  `json:"no_store,omitempty" structs:"no_store,omitempty"`
```
Comment (jefferai) [inline builtin/logical/pki/path_roles.go]: This doesn't need to be a `*bool`, just `bool` is fine, with associated updates elsewhere.

**Comment by slingamn** [comment]
Need anything else from me on this? (Sorry, I'm new to this CR UI.)

File: builtin/logical/pki/path_issue_sign.go [hunk 3]
```diff
@@ -240,10 +240,12 @@ func (b *backend) pathIssueSignCert(
...
-		Key:   "certs/" + cb.SerialNumber,
-		Value: parsedBundle.CertificateBytes,
-	})
+	if !role.NoStore {
+		err = req.Storage.Put(&logical.StorageEntry{
+			Key:   "certs/" + cb.SerialNumber,
+			Value: parsedBundle.CertificateBytes,
+		})
+	}
 	if err != nil {
```
Comment (chrishoffman) [inline builtin/logical/pki/path_issue_sign.go]: Can you move this error handling into the previous block?

**Comment by jefferai** [comment]
LGTM, thanks!

**Pull request merged**